unstructured_inference/models/largemodel.py

```diff
@@ -0,0 +1,171 @@
+import os
+from typing import List, Optional
+
+import numpy as np
+import torch
+from PIL.Image import Image
+from transformers import (
+    AutoTokenizer,
+    DonutImageProcessor,
+    DonutProcessor,
+    VisionEncoderDecoderModel,
+)
+
+from unstructured_inference.inference.layoutelement import LocationlessLayoutElement
+from unstructured_inference.models.unstructuredmodel import UnstructuredElementExtractionModel
+
+MODEL_TYPES = {
+    "large_model": {
+        "tokenizer_name": "xlm-roberta-large",
+        "pre_trained_model_name": "unstructuredio/ved-fine-tuning",
+    },
+}
+
+LABEL_MAP = [
+    "Title",
+    "Abstract",
+    "Headline",
+    "Subheadline",
+    "Text",
+    "Table",
+    "List",
+    "List-item",
+    "Page number",
+    "Header",
+    "Footer",
+    "Address",
+    "Author",
+    "Chart",
+    "Caption",
+    "Formula",
+    "Picture",
+    "Advertisement",
+    "Link",
+    "Misc",
+    "Field-Name",
+    "Value",
+    "Threading",
+    "Metadata",
+    "Form",
+]
+
+# NOTE(alan): Order matters with these
+SPECIAL_TAGS = ["<sep/>"] + [
+    f"<{open_or_close}s_{label_type}>" for label_type in LABEL_MAP for open_or_close in ("", "/")
+]
+
+
+class UnstructuredLargeModel(UnstructuredElementExtractionModel):
+    required_w: int = 1248
+    required_h: int = 1664
+
+    def initialize(
+        self,
+        tokenizer_name: str,
+        pre_trained_model_name: str,
+        auth_token: Optional[str] = os.environ.get("UNSTRUCTURED_HF_TOKEN"),
+    ):
+        """Load the model for inference."""
+        self.tokenizer: AutoTokenizer = AutoTokenizer.from_pretrained(tokenizer_name)
+
+        self.processor: DonutProcessor = DonutProcessor(
+            image_processor=DonutImageProcessor(
+                do_resize=True,
+                size=(self.required_w, self.required_h),
+            ),
+            tokenizer=self.tokenizer,
+        )
+
+        self.tokenizer.add_tokens(new_tokens=SPECIAL_TAGS, special_tokens=True)
+
+        self.model = VisionEncoderDecoderModel.from_pretrained(
+            pre_trained_model_name,
+            ignore_mismatched_sizes=True,
+            use_auth_token=auth_token,
+        )
+
+    def predict(self, image) -> List[LocationlessLayoutElement]:
+        """Do inference using the wrapped model."""
+        tokens = self.predict_tokens(image)
+        elements = self.postprocess(tokens)
+        return elements
+
+    def predict_tokens(self, image: Image) -> List[int]:
+        """Predict tokens from image."""
+        annotation = self.model.generate(
+            self.processor(
+                np.array(
+                    image,
+                    np.float32,
+                ),
+                return_tensors="pt",
+            ).pixel_values,
+            decoder_input_ids=torch.tensor([[0]]),
+            do_sample=True,
+            top_p=0.92,
```

Comment:
Do any of these settings need to change by model? For instance between the large model and donut?

Reply:
Good question. @ajjimeno @LaverdeS @benjats07 ?